README.md

```diff
@@ -32,9 +32,11 @@ See [Deployment Models](#deployment-models) for more information about the diffe
 
 The inputs to DATASIM consist of four parts, each represented by JSON. They are as follows:
 
-#### xAPI Profile
+#### xAPI Profile(s)
 
-A valid xAPI Profile is required for DATASIM to generate xAPI Statements. You can learn more about the xAPI Profile Specification [here](https://github.com/adlnet/xapi-profiles). This input can either be a single Profile JSON-LD document or an array of JSON-LD format profiles. At this time all referenced concepts in a Profile must be included in the input. For instance if in "Profile A" I have a Pattern that references a Statement Template found in "Profile B", both Profiles must be included in an array as the Profile input.
+One or more valid xAPI Profiles are required for DATASIM to generate xAPI Statements. You can learn more about the xAPI Profile Specification [here](https://github.com/adlnet/xapi-profiles). This input can either be a single Profile JSON-LD document or an array of JSON-LD format profiles. At this time all referenced concepts in a Profile must be included in the input. For instance if in "Profile A" I have a Pattern that references a Statement Template found in "Profile B", both Profiles must be included in an array as the Profile input.
+
+Note that by default, any patterns with a `primary` property set to `true` in the provided profiles will be used for generation. You can control which profiles these primary patterns are sourced from with the `gen-profiles` option by supplying one or more profile IDs. You can further control which specific primary patterns are used with the `gen-patterns` option by supplying one or more pattern IDs.
```

Comment:
What is the behavior of `gen-profiles` if any of the profiles listed aren't in the profile cosmos? (And likewise for `gen-patterns` and non-present patterns.)

Reply:
Don't know, I'll check it out. I believe it will hit an assertion error when the IRI-map has no primaries, which would be fine.

Reply:
Yeah, so the assert gets it on gen:

    Execution error (AssertionError) at com.yetanalytics.datasim.xapi.profile/pattern-zip (profile.clj:70).
    Assert failed: Profile input must contain primary patterns!
    (seq primary-pattern-iris)

BUT `validate-input` doesn't get it, so I'll add a cross-validation to `input` about it that checks the profiles + these args.

Reply:
Indeed to the average user that AssertionError message would be rather cryptic (also it doesn't show when the assert flag is off right?)

Reply:
yeah, but more to the point `validate-input` should validate the input, lol

Reply:
https://github.com/yetanalytics/datasim/pull/79/commits/9b006e0a631ca5d73dbe571b178d5ba0b2e40f18

Reply:
````shell
➜  datasim git:(DS-129) ✗ clojure -Mcli -m com.yetanalytics.datasim.main \
-p dev-resources/profiles/tla/mom.jsonld \
-a dev-resources/personae/simple.json \
-l dev-resources/alignments/simple.json \
-o dev-resources/parameters/simple.json \
--gen-profile http://example.com/foo \
validate-input
================================================================================
INPUT ERROR: parameters-gen-profiles-0
================================================================================
path: [:parameters :gen-profiles 0]

Profile ID http://example.com/foo is not one of provided profiles: https://w3id.org/xapi/tla
➜  datasim git:(DS-129) ✗ clojure -Mcli -m com.yetanalytics.datasim.main \
-p dev-resources/profiles/tla/mom.jsonld \
-a dev-resources/personae/simple.json \
-l dev-resources/alignments/simple.json \
-o dev-resources/parameters/simple.json \
--gen-pattern http://example.com/foo \
validate-input
================================================================================
INPUT ERROR: parameters-gen-patterns-0
================================================================================
path: [:parameters :gen-patterns 0]

Pattern ID http://example.com/foo is not among primary patterns in provided profiles: https://w3id.org/xapi/tla#completed_session

````

Reply:
Excellent.